dockstore-webservice/src/main/java/io/dockstore/webservice/resources/OrganisationResource.java

```diff
@@ -77,20 +77,24 @@ public OrganisationResource(SessionFactory sessionFactory) {
     @Timed
     @UnitOfWork
     @RolesAllowed({ "curator", "admin" })
-    @Path("/approve/{organisationId}")
+    @Path("{organisationId}/approve/")
     @ApiOperation(value = "Approves an organisation.", authorizations = { @Authorization(value = JWT_SECURITY_DEFINITION_NAME) }, notes = "Admin/curator only", response = Organisation.class)
     public Organisation approveOrganisation(@ApiParam(hidden = true) @Auth User user,
             @ApiParam(value = "Organisation ID.", required = true) @PathParam("organisationId") Long id) {
         Organisation organisation = organisationDAO.findById(id);
         if (organisation == null) {
             String msg = "Organisation not found";
             LOG.info(msg);
-            throw new CustomWebApplicationException(msg, HttpStatus.SC_BAD_REQUEST);
+            throw new CustomWebApplicationException(msg, HttpStatus.SC_NOT_FOUND);
         }
         organisation.setApproved(true);
 
-        Event createOrganisationEvent = new Event(null, organisation, null, null, user, Event.EventType.APPROVE_ORG);
-        eventDAO.create(createOrganisationEvent);
+        Event approveOrgEvent = new Event.Builder()
```

Comment:
Do you want to fire the event if the org was already approved? I think so, but just double-checking.

Reply:
I guess that could be done in this case, I'll add a check so that this isn't duplicated.